havlak/havlak.d

```diff
@@ -356,64 +364,67 @@ int findHavlakLoops(CFG cfg)
     return findHavlakLoops(cfg, lsg);
 }
 
-class LoopTesterApp {
+final class LoopTesterApp {
   CFG cfg;
   LSG lsg;
 
-  this() {
+  this() @safe {
     cfg = new CFG();
     lsg = new LSG();
   }
 
-  int buildDiamond(int start) {
-    int bb0 = start;
-    new BasicBlockEdge(cfg, bb0, bb0 + 1);
-    new BasicBlockEdge(cfg, bb0, bb0 + 2);
-    new BasicBlockEdge(cfg, bb0 + 1, bb0 + 3);
-    new BasicBlockEdge(cfg, bb0 + 2, bb0 + 3);
-    return bb0 + 3;
-  }
+  @safe pure
+  {
+    int buildDiamond(int start) {
+        int bb0 = start;
+        BasicBlockEdge(cfg, bb0, bb0 + 1);
+        BasicBlockEdge(cfg, bb0, bb0 + 2);
+        BasicBlockEdge(cfg, bb0 + 1, bb0 + 3);
+        BasicBlockEdge(cfg, bb0 + 2, bb0 + 3);
+        return bb0 + 3;
+    }
 
-  void buildConnect(int _start, int _end) {
-    new BasicBlockEdge(cfg, _start, _end);
-  }
+    void buildConnect(int _start, int _end) {
+        BasicBlockEdge(cfg, _start, _end);
```

Comment:
Just an indentation caused by:
+  @safe pure
+  {

removed unneeded new because BasicBlockEdge is a struct